components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx

```diff
@@ -2,21 +2,20 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import {localizeMessage} from 'utils/utils';
 
 import Menu from 'components/widgets/menu/menu';
 
-export default class CloseChannel extends React.PureComponent {
-    static propTypes = {
-        isArchived: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            goToLastViewedChannel: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    isArchived: boolean;
+    actions: {
+        goToLastViewedChannel: () => () => void;
```

Comment:
Does this need to be a function that returns a function? Is just `() => void` not enough here?

Reply:
well, yes, I think that in practical terms it is enough; although the goToLastViewChannel is a function that returns a function. I am not sure if there is a convention for this. I am happy to modify it to follow the MM standard.

Reply:
If I understand correctly, once the function is bound using a redux connector, the return value of `goToLastViewedChannel` would be the return value of the function it returns, since we aren't passing in `dispatch` or `getState` when calling it here. @hmhealey would know for sure.

Reply:
Yeah, that's correct. `bindActionCreators` in the connnector takes the `(arguments) => (dispatch, getState) => returns` that makes up a redux-thunk action like `goToLastViewedChannel`, and it turns it into `(arguments) => returns`. In this case, the type just ends up as `() => void`